### PR TITLE
Web access

### PR DESCRIPTION
**Pull request by kog**
Add a new `WebAcces` module that generates URLs with tokens in them, allowing folks on IRC to log into the MVC UI without a username/password combo

### CODE REVIEW

File: src/SpikeCore/SpikeCore.Web/Controllers/AuthenticationController.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            _userManager = userManager;
+        }
+
+        [HttpGet]
+        public async Task<IActionResult> TokenCallback(string email, string token)
+        {
+            var user = await _userManager.FindByEmailAsync(email);
+
+            var isValid =
+                await _userManager.VerifyUserTokenAsync(user, "PasswordlessLoginProvider", "passwordless-auth", token);
```
Comment (Suchiman) [inline src/SpikeCore/SpikeCore.Web/Controllers/AuthenticationController.cs:35]: Would be nice to either have const's or `nameof(PasswordlessLoginTokenProvider)`, although interestingly enough, nameof on a generic type needs an type argument that it then proceeds to ignore, e.g. `nameof(Meh<int>)` -> `"Meh"`

File: src/SpikeCore/SpikeCore.Web/Controllers/AuthenticationController.cs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+                // This prevents replay attacks by swapping out the security stamp. This works because all  tokens
+                // contain the security stamp that was used to generate them.
+                await _userManager.UpdateSecurityStampAsync(user);
+
+                await HttpContext.SignInAsync(
+                    IdentityConstants.ApplicationScheme,
+                    new ClaimsPrincipal(
+                        new ClaimsIdentity(new List<Claim> {new Claim("sub", user.Id)},
+                            IdentityConstants.ApplicationScheme)
+                    ));
```
Comment (Suchiman) [inline src/SpikeCore/SpikeCore.Web/Controllers/AuthenticationController.cs:48]: what about `_signInManager.SignInAsync(user)`?

Comment (kog) [reply src/SpikeCore/SpikeCore.Web/Controllers/AuthenticationController.cs:48]: Well, I mean... that wasn't in the article I blatantly copied and pasted. But now that you mention it, that's a smarter idea... And this is the benefit of code reviews :D

File: src/SpikeCore/SpikeCore.Web/Controllers/AuthenticationController.cs (hunk 1 above)
Comment (kog) [reply src/SpikeCore/SpikeCore.Web/Controllers/AuthenticationController.cs:35]: After talking about this on IRC, I'll go ahead and drop the generics on the type... we really only have one type of user anyway.

File: src/SpikeCore/SpikeCore.Web/Controllers/AuthenticationController.cs (hunk 2 above)
Comment (kog) [reply src/SpikeCore/SpikeCore.Web/Controllers/AuthenticationController.cs:48]: That being said, I'm wondering if they're probably supporting a more complicated environment. All we're doing here is basic role checking...

Comment (Suchiman) [reply src/SpikeCore/SpikeCore.Web/Controllers/AuthenticationController.cs:48]: signInManager takes care of automatically putting claims and roles associated with the user into the `ClaimsIdentity`. My guess would be that they either didn't knew or simplified it for the sake of example... or copied it from elsewhere :trollface:

Comment (kog) [reply src/SpikeCore/SpikeCore.Web/Controllers/AuthenticationController.cs:48]: Yeah, but who'd be dumb enough to copy and paste things from the internet? Uh, asking for a friend...

**Pull request merged**